### PR TITLE
fix(tests): fixing providers integration tests compilation

### PR DESCRIPTION
**Pull request by geekbrother**
# Description

This PR fixes RPC providers [integration test compilation errors](https://github.com/WalletConnect/blockchain-api/actions/runs/9548607111/job/26317101711#step:4:1099) after removing the rudimental code in #673.

## How Has This Been Tested?

* Passed `cargo test coinbase_provider -- --ignored` locally,
* Passed `cargo clippy --workspace --all-features --all-targets -- -D warnings` locally.

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: tests/context/mod.rs [hunk 1]
```diff
@@ -23,8 +23,6 @@ impl AsyncTestContext for ServerContext {
             {
                 let project_id = env::var("PROJECT_ID").expect("PROJECT_ID must be set");
                 RpcProxy {
-                    private_addr: None,
```
Comment (geekbrother) [inline tests/context/mod.rs:26]: Initialization was removed in #673

File: tests/context/server.rs [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#[cfg(feature = "test-localhost")]
```
Comment (chris13524) [inline tests/context/server.rs:1]: Why is this change required?

Comment (geekbrother) [reply tests/context/server.rs:1]: This fixing clippy warning when `test *_provider -- --ignored` invoked.
